Simplenote/Classes/SPNoteEditorViewController.m

```diff
@@ -682,11 +680,14 @@ - (void)scrollViewDidScroll:(UIScrollView *)scrollView
 // only called for user changes
 // safe to alter text attributes here
 
-- (BOOL)textViewShouldBeginEditing:(UITextView *)textView {
-    
-    if (self.searching)
-        [self endSearching:textView];
-    
+- (BOOL)textViewShouldBeginEditing:(UITextView *)textView
+{
+    if ([self isShowingHistory]) {
+        return NO;
+    }
+
+    [self ensureSearchIsDismissed];
```

Comment:
Is it possible this `ensureSearchIsDismissed` invocation came thru a bad merge? can we drop it? (since it's a different behavior)

Reply:
Dismissing search (bottom bar and highlights) is a current behaviour when starting editing a note while searching. I think the main goal is to remove the highlights and let the user edit text without distractions.
Same code from develop:
https://github.com/Automattic/simplenote-ios/blob/16853f8fde4a20595edba4c0e8e1b32ccf45ac94/Simplenote/Classes/SPNoteEditorViewController.m#L860

I extracted this code to `ensureSearchIsDismissed` method so that we can use it from `showHistoryController` as well.

Reply:
Woooo I tried this yesterday (AppStore) and I was entirely sure it remained in search mode. I'll take another look, apologies about the noise!!